File: kuba-front/src/app/shared/models/incident-update-request.ts
import IncidentType from "../enums/incident-type.enum"
import Severity from "../enums/severity.enum"

export default interface IncidentUpdateRequest {
    title?: string
    description?: string
    severityType?: Severity
    incidentType?: IncidentType
    reportedDate?: Date
    updatedBy: string
}

File: kuba-back/Kuba.Api/Kuba.Api/Dtos/Incident/IncidentUpdateRequest.cs
using Kuba.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace Kuba.Api.Dtos.Incident
{
    public class IncidentUpdateRequest
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime? ReportedDate { get; set; }
        public SeverityType? SeverityType { get; set; }
        public IncidentType? IncidentType { get; set; }
        [Required]
        public Guid UpdatedBy { get; set; }
    }
}
